test/jdk/java/util/zip/ZipInputStream/Zip64DataDescriptor.java

```diff
@@ -0,0 +1,275 @@
+     */
+    @Test
+    public void shouldIgnoreNoMagicMarkers() throws IOException {
+        // Set compressed and uncompressed size fields to zero
+        ByteBuffer.wrap(invalidZip64).order(ByteOrder.LITTLE_ENDIAN)
+                .putInt(ZipFile.LOCSIZ, 0)
+                .putInt(ZipFile.LOCLEN, 0);
+
+
+        readZipInputStream(invalidZip64);
+    }
+
+    /*
+     * Validate that an extra data size exceeding the length of the extra field is ignored
+     */
+    @Test
+    public void shouldIgnoreTrucatedZip64Extra() throws IOException {
+
+        truncateZip64();
+
+        readZipInputStream(invalidZip64);
+    }
+
+    /**
+     * Update the Extra field header ID of the invalid file
+     */
+    private void setExtraHeaderId(short id) {
+        // Set the header ID on the extra field
+        ByteBuffer buffer = ByteBuffer.wrap(invalidZip64).order(ByteOrder.LITTLE_ENDIAN);
+        int nlen = buffer.getShort(ZipFile.LOCNAM);
+        buffer.putShort(ZipFile.LOCHDR + nlen, id);
+    }
+
+    /**
+     * Updates the 16-bit 'data size' field of the Zip64 extended information field,
+     * potentially to an invalid value.
+     * @param size the value to set in the 'data size' field.
+     */
+    private void setExtraSize(short size) {
+        ByteBuffer buffer = ByteBuffer.wrap(invalidZip64).order(ByteOrder.LITTLE_ENDIAN);
+        // Compute the offset to the Zip64 data block size field
+        short nlen = buffer.getShort(ZipFile.LOCNAM);
+        int dataSizeOffset = ZipFile.LOCHDR + nlen + Short.BYTES;
+        buffer.putShort(dataSizeOffset, size);
+    }
+
+    /**
+     * Puts a truncated Zip64 field (just the tag) at the end of the LOC extra field.
+     * The beginning of the extra field is filled with a generic extra field containing
+     * just zeros.
+     */
+    private void truncateZip64() {
+        ByteBuffer buffer = ByteBuffer.wrap(invalidZip64).order(ByteOrder.LITTLE_ENDIAN);
+        // Get the LOC name and extra sizes
+        short nlen = buffer.getShort(ZipFile.LOCNAM);
+        short elen = buffer.getShort(ZipFile.LOCEXT);
+        int cenOffset = ZipFile.LOCHDR + nlen + elen;
+
+        // Zero out the extra field
+        int estart = ZipFile.LOCHDR + nlen;
+        buffer.put(estart, new byte[elen]);
+        // Put a generic extra field in the start
+        buffer.putShort(estart, (short) 42);
+        buffer.putShort(estart + Short.BYTES, (short) (elen - 4 - 2));
+        // Put a truncated (just the tag) Zip64 field at the end
+        buffer.putShort(cenOffset - Short.BYTES, (short) 0x0001);
+    }
+
+    /*
+     * Consume all entries in a ZipInputStream, possibly throwing a
+     * ZipException if the reading of any entry stream fails.
+     */
+    private void readZipInputStream(byte[] zip) throws IOException {
+        try (ZipInputStream in = new ZipInputStream(new ByteArrayInputStream(zip))) {
+            ZipEntry e;
+            while ( (e = in.getNextEntry()) != null) {
```

Comment:
The zip is expected to have a single ZipEntry. Would it be better to do something like this, so that if the ZipEntry is missing, then the test fails?

(I haven't tested this code below)

```java
ZipEntry e = in.getNextEntry();
assertNotNull(e, "missing zip entry");
assertEquals("hello\n", new String(in.readAllBytes(), StandardCharsets.UTF_8));
assertNull(in.getNextEntry(), "unexpected additional zip entry");

```